ocs-api/import-owner-private-keys2.sh

```diff
@@ -0,0 +1,300 @@
+      rm owner-public-key.pem
+    fi
+    #adding delimiters for SDO 1.9 pub key format
+    echo "," >> rsapub-key.pem && echo "," >> ecdsa256pub-key.pem
+    echo "Concatenating Public Key files..."
+    for i in "rsapub-key.pem" "ecdsa256pub-key.pem" "ecdsa384pub-key.pem"
+      do
+        local keyName=$i
+        local removeWord="pub-key.pem"
+        keyName=${keyName//$removeWord/}
+        #adding delimiters for SDO 1.9 pub key format
+        echo ""$keyName":" >key.txt
+        #make sure key name is lowercase for file name
+        cat key.txt $i >> ${ORG_UNIT}_${KEY_NAME}_public-key.pem
+      done
+    chk $? 'Concatenating Public Key files...'
+    rm -- ecdsa*.pem && rm rsapub* && rm key.txt
+  fi
+  mv ${ORG_UNIT}_${KEY_NAME}_public-key.pem ocs/config/db/v1/creds/publicKeys
+  cd ocs/config/db/v1/creds/publicKeys
+  if [ ! -d "${ORG_UNIT}" ]; then
+    mkdir ${ORG_UNIT} && mv ${ORG_UNIT}_${KEY_NAME}_public-key.pem ${ORG_UNIT}/
+  else 
+    mv ${ORG_UNIT}_${KEY_NAME}_public-key.pem ${ORG_UNIT}/
+  fi
+  cd
+}
+
+#This function will take the private key and certificate that has been passed in, and use them to generate a keystore pkcs12 file containing both files.
+function genKeyStore(){
+  #echo $SDO_KEY_PWD
+  for i in "rsa" "ecdsa256" "ecdsa384"
+      do
+        # Convert the keyCertificate and private key into ‘PKCS12’ keystore format:
+        cd "$i"Key/ && openssl pkcs12 -export -in "$i"Cert.crt -inkey "$i"private-key.pem -name "${ORG_UNIT}_${KEY_NAME}_$i" -out "${KEY_NAME}_$i.p12" -password pass:"$SDO_KEY_PWD"
+        chk $? 'Converting private key and cert into keystore'
+        cp "${KEY_NAME}_$i.p12" .. && rm -- *
+        cd .. && rmdir "$i"Key
+      done
+}
+
+function insertKeys(){
+  #This function will insert all private keystores into the master keystore
+  if [[ -f "${KEY_NAME}_$i.p12" ]]; then
+    for i in "rsa" "ecdsa256" "ecdsa384"
+      do
+        #Import custom keystores into the master keystore. /usr/lib/jvm/openjre-11-manual-installation/bin/
+        echo "yes" | /usr/lib/jvm/openjre-11-manual-installation/bin/keytool -importkeystore -destkeystore ocs/config/db/v1/creds/owner-keystore.p12 -deststorepass "$SDO_KEY_PWD" -srckeystore "${KEY_NAME}_$i.p12" -srcstorepass "$SDO_KEY_PWD" -srcstoretype PKCS12 -alias "${ORG_UNIT}_${KEY_NAME}_$i"
+        chk $? "Inserting "${KEY_NAME}_$i.p12" keystore into ocs/config/db/v1/creds/owner-keystore.p12"
+      done
+    rm -- *.p12
+  else
+    echo "One or more of the keystores are missing. There should be three keystores of type rsa, ecdsa256, and ecdsa384"
+    exit 2
+fi
+}
+
+
+#============================MAIN CODE=================================
+
+ensureWeAreUser
+getKeyPair
+
+if [[ -n "$keyType" ]] && [[ "$keyType" == "all" ]]; then
+  allKeys
+  combineKeys
+else
+  genKey
+fi
+echo "Owner Private Keys and Owner Public Key have been created"
+
+genKeyStore
+insertKeys
+echo "Owner private key pairs have been imported."
+
```

Comment:
2 overall comments for this script:
- you should use different exit codes for different errors, so in your golang code you can return a simple summary msg and then the detailed errors msgs from the script
- have you thought about what happens if 2 clients run this api at exactly the same time? Will they collide on some of the temporary files you use? The file and dir names that are specific to the org and key name should be ok, but for the temporary files that aren't, a good practice is do `TMP_DIR=$(mktemp -d)` at the beginning, then prepend `$TMP_DIR/` to all of your temporary files. Mktemp creates a unique directory under `/tmp`, which has the added advantage that if your script forgets to clean up some of the temp files, the OS will eventually get rid of them.